### DISCUSSION TITLE
Is there any existing method to detect distorted structures in rdkit?

### DISCUSSION THREAD

**Discussion by HelloJocelynLu** [rdkit/rdkit#7087] [Q&A] [answered]
Suppose I have a bunch of molecules in SDF format. Are there any ways to detect whether these molecules are distorted? For example, in terms of bond lengths or angles.
Please see an example of distorted structure below:
![image](https://github.com/rdkit/rdkit/assets/34988596/1d15d02b-ac7a-4545-959b-5c4e908ce909)
Thank you for any suggestions!

**Comment by LiuCMU** [comment]
Not sure how you define the distorted molecules. If you want to find conformers with clashing atoms, you could get the atomic position as an array, then computing the minimal pairwise distance. If the minimal distance is below 0.7 angstrom, then there must be clashing atoms.

**Reply by HelloJocelynLu** [reply]
Hello, thank you for your response. Of course, clash is one of the situations. However, distorted molecules may also have abnormal bond lengths and bond angles. I am wondering if there are any existing methods for detecting those abnormalities.

**Reply by bp-kelley** [reply, marked as answer]
For bond lengths you could minimize the structure with MMFF and compare them for large differences.

**Reply by HelloJocelynLu** [reply]
That's a great idea. Thank you for your insight. I will try this one.
